### ISSUE TITLE
Inconsistent firstRun Default Behaviors

### ISSUE THREAD

**Issue by DistractionRectangle** [Ryochan7/DS4Windows#2963]
**Describe the bug**

During a `firstRun` on a clean system (where the directory described by [appDataPpath](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ScpUtil.cs#L518) does not exist), DS4Windows exhibits different defaults depending on whether "Program Folder" (portable mode) or "AppData" is chosen in the `SaveWhere` form. When saving config under "AppData" all controllers are supported by default; in portable mode, controller support defaults to disabled for everything but DS4 controllers. This deviates from the behavior in versions <= v3.1.12, where in either mode controller support defaults to enabled for all controller types supported by DS4Windows.

**To Reproduce**
1. Delete DS4Windows directory located at `appDataPpath`, on my system this is AppData\Roaming\DS4Windows
2. Download any release >= v3.2.0
3. On first run, select "Program Folder" and observe device support under Settings -> Device Options; alternatively observe Profiles.xml
4. Close out and delete portable version
5.  Download any release >= v3.2.0
6. On first run, select "AppData" and observe device support under Settings -> Device Options; alternatively observe Profiles.xml
7. Repeat steps 1-6 with a version <= v3.1.12

**Expected behavior**
DS4Windows has consistent and predictable defaults between versions and modes.

tl;dr "Program Folder" uses defaults set in `InputDeviceOptions` while "AppData" uses defaults set in `ControlServiceDeviceOptions` previous versions used the defaults set in `ControlServiceDeviceOptions`

<details>
  <summary>Click to Expand Long Version</summary>
  
**Screenshots and Logs**
![firstRun_using_AppData](https://github.com/Ryochan7/DS4Windows/assets/48034372/1fe2b06e-94c7-488d-87a2-243b15d96930)
    - firstRun using AppData as save location

![firstRun_using_Program_Folder](https://github.com/Ryochan7/DS4Windows/assets/48034372/429dba27-7ef6-4f40-8636-6fbfda29cccf)
    - firstRun using Program Folder as save location

**Additional context**
The cause is simple enough, it's that when the `appDataPpath` directory doesn't exist, it's not created until after the SaveWhere flow - it's a few lines later, [here](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/App.xaml.cs#L145). This means that when "AppData" is chosen as a the save location, the [SaveDefault](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ScpUtil.cs#L849) function fails to seed `Profiles.xml`. 

In turn, this means the subsequent call to [Load](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/App.xaml.cs#L167) fails, and does not invoke [MapTo()](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/DTOXml/AppSettingsDTO.cs#L833). 

This matters because we have two classes which can potentially seed `AppSettingsDTO.DeviceOptions.<controller>SupportSettings.Enabled`: the initial values from the [InputDeviceOptions class](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/DTOXml/AppSettingsDTO.cs#L947) and the [ControlServiceDeviceOptions class](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ControlServiceDeviceOptions.cs#L15). The values in InputDeviceOptions default to false [unless explicitly configured](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/DTOXml/AppSettingsDTO.cs#L996), while the values supplied by the ControlServiceDeviceOptions class all default to true [\[0\]](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ControlServiceDeviceOptions.cs#L64), [\[1\]](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ControlServiceDeviceOptions.cs#L180), [\[2\]](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ControlServiceDeviceOptions.cs#L308), [\[3\]](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ControlServiceDeviceOptions.cs#L407) expect for the newly added [DS3DeviceOptions class](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ControlServiceDeviceOptions.cs#L81)

[BackingStore.deviceOptions](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ScpUtil.cs#L3406) is initialized with the [ControlServiceDeviceOptions class](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ControlServiceDeviceOptions.cs#L15), unless overwritten by MapTo. MapTo uses the initial values from [AppSettingsDTO.DeviceOptions](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/DTOXml/AppSettingsDTO.cs#L734), which is initialized by the [InputDeviceOptions class](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/DTOXml/AppSettingsDTO.cs#L947), ***unless*** these values are overwritten [xml deserializer that loads Profiles.xml](https://github.com/Ryochan7/DS4Windows/blob/2eef61aa6f7d03b569a4708ac86ae0382aa475e0/DS4Windows/DS4Control/ScpUtil.cs#L7306) (which when seeded by SaveDefault, no values are present, and so we still use the initial values from InputDeviceOptions on firstRun).

</details>

Ignoring that we have two classes that seem to do the same thing, it's easy to fix. The question is, what the fix should actually look like.

- What should the defaults be?
    - If controller support is supposed to default to disabled, why? IMO, controller support is something one should opt out of rather than opt into. Since old versions behaved this way, I don't see the change it unless there's a compelling reason.
- What should the fix look like? There's a few different places where a fix can be applied
    - patching `SaveDefault` to invoke `MapTo` even when `Profiles.xml` doesnt exist
    - bubbling up `CreateConfDirSkeleton` to be invoked in the `SaveWhere` form before `SaveDefault`
    - reconciling `InputDeviceOptions` with `ControlServiceDeviceOptions`
    - etc
- Potentially fodder for a different issue/PR, but I think logging could be improved to communicate to the user when controllers are detected but not supported under Device Options. I bring it up because it's what has led me to look into this, and has caused confusion in issues here and on reddit.

**Comment by Ryochan7** [comment]
Thank you for catching this. I almost always use the "Program Folder" option and only test "AppData" on the rare occasion. I changed some of the classes to properly default to false and expose const values for outside use of those values.

https://github.com/Ryochan7/DS4Windows/commit/a186bdfc1d3fc16a1ba696d989fc12d80bcfb420

> If controller support is supposed to default to disabled, why? IMO, controller support is something one should opt out of rather than opt into. Since old versions behaved this way, I don't see the change it unless there's a compelling reason.

One reason was to not interfere with other mappers that people might want to use for other controllers; an example would be using BetterJoy for Switch Pro and JoyCon mapping rather than DS4Windows. It would also cause fewer problems with Steam Input.

> Potentially fodder for a different issue/PR, but I think logging could be improved to communicate to the user when controllers are detected but not supported under Device Options. I bring it up because it's what has led me to look into this, and has caused confusion in issues here and on reddit.

The closest option available right now is the `Enable verbose log` option in the Device Options window. I had never thought about putting something in the Log warning about possible usable controllers. Either way, I am not sure I would want to clutter the log with messages about potentially found controllers. Also that would require keeping a list of currently disabled controller file handles in the DS4Devices class so messages would only appear when DS4Windows first detects a controller rather than at every hotplug call.

**Comment by DistractionRectangle** [comment]
> One reason was to not interfere with other mappers that people might want to use for other controllers; an example would be using BetterJoy for Switch Pro and JoyCon mapping rather than DS4Windows. It would also cause fewer problems with Steam Input.

Both are something I didn't consider and are good reasons. Okay, so defaulting with most controller support disabled by default makes sense. I can spin up a PR to fix it this weekend.

> Either way, I am not sure I would want to clutter the log with messages about potentially found controllers. Also that would require keeping a list of currently disabled controller file handles in the DS4Devices class so messages would only appear when DS4Windows first detects a controller rather than at every hotplug call.

Yeah, that makes sense. Perhaps something more like a one time startup message `Support for <controllers affected> is currently disabled. Support can be toggled under Settings -> Device Options`.

**Comment by DistractionRectangle** [comment]
Ha, I missed you already pushed the changes when I read your response, Closing
